### ISSUE TITLE
Editor locked when opening scene with C# [Tool] script

### ISSUE THREAD

**Issue by Torguen** [godotengine/godot#34619]
**Godot version:**
3.2 beta 4 Mono

**OS/device including version:**
Win10 64 Home

**Issue description:**
The editor is partially blocked when the scene "src / levels / Level01.tscn" or "src / Objects / Portal2D.tscn" is opened.
The cmd console does not report anything

**Steps to reproduce:**
Import the project and open the scene indicated above.

**Minimal reproduction project:**
[plataformasTutorial.zip](https://github.com/godotengine/godot/files/4003092/plataformasTutorial.zip)

**Comment by Torguen** [comment]
If the Portal2d.tscn file is edited and the "Portal2d" script is deleted then it works.

**Comment by qarmin** [comment]
Seems like .mono/temp was broken.
Delete .mono folder and the problem will disappear.

**Comment by Torguen** [comment]
I have tried it but it has not solved it.

**Comment by Torguen** [comment]
It still happens in 3.2 beta 6 mono

**Comment by akien-mga** [comment]
I can confirm the original issue with the project in the OP, and also that deleting the `res://.mono` folder seems to fix it.

**Comment by Torguen** [comment]
No, I tried it, the editor will crash again when you reload the project.
For example when you do "Project" -> "Exit to the list of projects" then reload the project and the editor crashes again.

**Comment by akien-mga** [comment]
Indeed, after importing it to the project manager and editing it from there, I can reproduce the bug.

Interestingly, I also this when it's frozen:
![Screenshot_20200113_142525](https://user-images.githubusercontent.com/4701338/72259434-bc045a80-3610-11ea-89de-2d6684d2391a.png)

So all platforms seem to register themselves as one-click deploy for some reason.

**Comment by neikeq** [comment]
What's the real issue here? Is there a crash? Is the editor completely broken? Or is the editor responsive, but clicking stuff does nothing?

If it's the latter, then it's clear why it's happening. You're calling `GetTree().Paused = true;` in your tool script's `_Ready`. Same thing can be reproduced in GDScript. Just add the following script to a node in your scene:

```
tool
extends Node

func _ready():
	get_tree().paused = true
```

You may need to restart the editor for `_ready` to be called.

**Comment by neikeq** [comment]
If you delete the `res://.mono` folder and start the editor again, it doesn't happen because there's no code to run. Once you rebuild, it will start happening again.

**Comment by akien-mga** [comment]
Indeed, it's a `tool` script misuse then. I recently updated the docs to warn about the potential impact of code interacting with the scene tree in tool scripts: https://docs.godotengine.org/en/latest/tutorials/misc/running_code_in_the_editor.html

It doesn't explain the super weird screenshot I got in https://github.com/godotengine/godot/issues/34619#issuecomment-573662618, but I also couldn't reproduce it so it's not a big deal.

**Comment by Torguen** [comment]
So is TOOL being misused here too? I followed this tutorial.
[https://youtu.be/6ziIyx60N6I?t=2120](url)

**Comment by akien-mga** [comment]
> So is TOOL being misused here too? I followed this tutorial.
He did not get any errors and everything worked correctly, I got the error and reported it here.
>
> https://youtu.be/6ziIyx60N6I?t=2120

I checked and I don't see any use of `get_tree().paused = true` in that tutorial, which is what pauses the editor in your version.

On the other hand, it does mix editor and game logic in `Portal2D.gd` in a way which could lead to unwanted behavior in the `teleport()` function: https://github.com/GDQuest/godot-beginner-2d-platformer/blob/master/game/src/Objects/Portal2D.gd#L21
If you were to move a PhysicsBody2D over it while in the editor, it would likely attempt to play the animation and change the scene which is being edited, which *could* lead to a crash too. (CC @NathanLovato FYI)
